Snowflake.Data/Core/SFStatement.cs

```diff
@@ -182,6 +185,7 @@ private SFBaseResultSet BuildResultSet(QueryExecResponse response, CancellationT
         {
             if (response.success)
             {
+                _lastQueryId = response.data.queryId;

```

Comment:
what about non successful scenario, do we happen to have queryId as well and can provide even though result set might not have any rows?

Reply:
@sfc-gh-ext-simba-hx @sfc-gh-igarish if it makes sense let's provide queryId in the unhappy path as well. @sfc-gh-pfus said it provided by the server in such a case as well.

Reply:
I checked on JDBC, ODBC, NodeJS and queryId is provided in those when there is a result set returned. So I would do a similar thing here (resultset might contain just error message) and provide queryId not only for reponse.success

Reply:
Posted more details through slack. For my understanding JDBC/ODBC throw exception when query fails (by checking "success" field in query response) therefore only set queryId on success case.

Reply:
done